cloudevents/extensions/bam.md

```diff
@@ -0,0 +1,96 @@
+# Business Activity Monitoring (BAM) Extension
+
+Business Activity Monitoring (BAM) was originally coined by analysts at Gartner,
+and refers to the aggregation, analysis, and presentation of real-time information
+about activities inside organizations, customers, and partners.
+
+This extension defines attributes that can be included within a CloudEvent
+to describe the business activity that the event is associated with.
+
+Produces and consumers are free to define an out-of-band agreement on the
+semantic meaning, or valid values, for the attribute.
+
+## Notational Conventions
+
+As with the main [CloudEvents specification](../spec.md), the key words "MUST",
+"MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+However, the scope of these key words is limited to when this extension is
+used. For example, an attribute being marked as "REQUIRED" does not mean
+it needs to be in all CloudEvents, rather it needs to be included only when
+this extension is being used.
+
+## Attributes
+
+### businessTransactionId
+
+- Type: `String`
+- Description: A unique identifier for the transaction.
+- Constraints
+  - REQUIRED
+  - MUST be a non-empty string
+  - RECOMMENDED as monotonically increasing and contiguous identifier that is lexicographically-sortable. 
+
+### businessProcessId
+
+- Type: `String`
+- Description: A unique identifier for the business process associated with the event.
+- Constraints
+  - REQUIRED
+  - MUST be a non-empty string
+  - RECOMMENDED a alphanumeric string that contains non-whitespace characters and only hyphens, underscores, and periods.
+
+### businessProcessTransactionId
+
+- Type: `String`
+- Description: A unique identifier for the business process transaction associated with the event.
+- Constraints
+  - REQUIRED
+  - MUST be a non-empty string
+  - RECOMMENDED a alphanumeric string that contains non-whitespace characters and only hyphens, underscores, and periods.
+
+### businessProcessTransactionStepId
+
+- Type: `String`
+- Description: A unique identifier for the business process transaction step associated with the event.
+- Constraints
+  - REQUIRED
+  - MUST be a non-empty string
+  - RECOMMENDED a alphanumeric string that contains non-whitespace characters and only hyphens, underscores, and periods.
+
+### businessProcessTransactionStepStatus
+
+- Type: `String`
+- Description: The status of the business process transaction step associated with the event.
+- Constraints
+  - OPTIONAL
+  - MUST be a non-empty string
```

Comment:
minor thing, add "if present, " to the start of lines like this where the attribute is OPTIONAL